pvnet/datamodule.py

```diff
@@ -16,86 +17,7 @@ def collate_fn(samples: list[NumpySample]) -> TensorBatch:
-        sample_class: sample class type to use for save/load/to_numpy
-    """
-
-    def __init__(self, sample_dir: str, sample_class: SampleBase):
-        """Initialise PresavedSamplesDataset"""
-        self.sample_paths = glob(f"{sample_dir}/*")
-        self.sample_class = sample_class
-
-    def __len__(self) -> int:
-        return len(self.sample_paths)
-
-    def __getitem__(self, idx) -> NumpySample:
-        sample = self.sample_class.load(self.sample_paths[idx])
-        return sample.to_numpy()
-
-
-class BasePresavedDataModule(LightningDataModule):
-    """Base Datamodule for loading pre-saved samples."""
-
-    def __init__(
-        self,
-        sample_dir: str,
-        batch_size: int = 16,
-        num_workers: int = 0,
-        prefetch_factor: int | None = None,
-        persistent_workers: bool = False,
-        pin_memory: bool = False,
-    ):
-        """Base Datamodule for loading pre-saved samples
-
-        Args:
-            sample_dir: Path to the directory of pre-saved samples.
-            batch_size: Batch size.
-            num_workers: Number of workers to use in multiprocess batch loading.
-            prefetch_factor: Number of batches loaded in advance by each worker.
-            persistent_workers: If True, the data loader will not shut down the worker processes 
-                after a dataset has been consumed once. This allows to maintain the workers Dataset 
-                instances alive.
-            pin_memory: If True, the data loader will copy Tensors into device/CUDA pinned memory 
-                before returning them.
-        """
-        super().__init__()
-
-        self.sample_dir = sample_dir
-
-        self._common_dataloader_kwargs = dict(
-            batch_size=batch_size,
-            sampler=None,
-            batch_sampler=None,
-            num_workers=num_workers,
-            collate_fn=collate_fn,
-            pin_memory=pin_memory,
-            drop_last=False,
-            timeout=0,
-            worker_init_fn=None,
-            prefetch_factor=prefetch_factor,
-            persistent_workers=persistent_workers,
-            multiprocessing_context="spawn" if num_workers>0 else None,
-        )
-
-    def _get_premade_samples_dataset(self, subdir: str) -> Dataset:
-        raise NotImplementedError
-
-    def train_dataloader(self) -> DataLoader:
-        """Construct train dataloader"""
-        dataset = self._get_premade_samples_dataset("train")
-        return DataLoader(dataset, shuffle=True, **self._common_dataloader_kwargs)
-
-    def val_dataloader(self) -> DataLoader:
-        """Construct val dataloader"""
-        dataset = self._get_premade_samples_dataset("val")
-        return DataLoader(dataset, shuffle=False, **self._common_dataloader_kwargs)
-
-
-class BaseStreamedDataModule(LightningDataModule):
+class BaseDataModule(LightningDataModule):
     """Base Datamodule which streams samples using a sampler for ocf-data-sampler."""
```

Comment:
Small typo we can tidy I think 
```suggestion
    """Base Datamodule which streams samples using a sampler from ocf-data-sampler."""
```